go-runtime/encoding/encoding.go

```diff
@@ -31,37 +27,29 @@ func encodeValue(v reflect.Value, w *bytes.Buffer) error {
 		w.WriteString("null")
 		return nil
 	}
+
 	t := v.Type()
-	switch {
-	case t.Kind() == reflect.Ptr && t.Elem().Implements(jsonMarshaler):
-		v = v.Elem()
-		fallthrough
 
-	case t.Implements(jsonMarshaler):
-		enc := v.Interface().(json.Marshaler) //nolint:forcetypeassert
-		data, err := enc.MarshalJSON()
+	// Special-cased types
+	switch {
+	case t == reflect.TypeFor[time.Time]():
+		data, err := json.Marshal(v.Interface().(time.Time))
 		if err != nil {
 			return err
 		}
 		w.Write(data)
 		return nil
 
-	case t.Kind() == reflect.Ptr && t.Elem().Implements(textMarshaler):
-		v = v.Elem()
-		fallthrough
-
-	case t.Implements(textMarshaler):
-		enc := v.Interface().(encoding.TextMarshaler) //nolint:forcetypeassert
-		data, err := enc.MarshalText()
-		if err != nil {
-			return err
-		}
-		data, err = json.Marshal(string(data))
+	case t == reflect.TypeFor[json.RawMessage]():
```

Comment:
We use it in ingress: https://github.com/TBD54566975/ftl/blob/main/backend/controller/ingress/response.go#L19

Without this, the `omitempty` behavior doesn't happen, and [this Ingress test](https://github.com/TBD54566975/ftl/blob/main/backend/controller/ingress/handler_test.go#L24C6-L24C17) fails because `"error": ""` gets included in the response body, which also breaks the [status code](https://github.com/TBD54566975/ftl/blob/main/backend/controller/ingress/handler_test.go#L107).

That's a really good point... should I update encoding to handle the `omitempty` next, so that this special case for `json.RawMessage` can be removed?